### ISSUE TITLE
Show JavaScript template js_nocookie only if an authenticated user exists or the session has been started

### ISSUE THREAD

**Issue by Mynyx** [contao/contao#1636]
**Affected version(s)**

Contao 4.9

**Description**

The JavaScript template ``js_nocookie`` shows a warning in forms, if cookies are disabled in the browser. It should prevent errors at the CSRF check (https://github.com/contao/contao/issues/134).

But from version 4.8 the CSRF check normaly is only performed if any cookies exist. Therefore the warning would be misleading and normaly not necessary.

It is only required and should only be shown if an authenticated user exists (HTTP authentication) or the session has been started (session with URI parameter).

https://github.com/contao/contao/blob/bf87609767e5623d9da8b98920683bef2d651997/core-bundle/src/EventListener/RequestTokenListener.php#L71-L75

Maybe the template could alternatively be dropped completely as these are now only exceptional cases.

**Comment by leofeyer** [comment]
@ausi @Toflar /cc

**Comment by Toflar** [comment]
> It is only required and should only be shown if an authenticated user exists (HTTP authentication) or the session has been started (session with URI parameter).

That's correct, although I don't know what you mean by URI parameter. It's required for basic auth (any `Authorization` header) or if any cookie is set.
We can't check if cookies are set in JavaScript though because of httponly cookies (such as the php session).

The only reliable method would be an ajax request to determine if checking the token is required.
On the other hand, showing this message to non-authorized users isn't really a problem, if you ask me...

**Comment by leofeyer** [comment]
> On the other hand, showing this message to non-authorized users isn't really a problem, if you ask me...

Agreed.
